modules/repository-function/main.tf

```diff
@@ -49,4 +49,14 @@ resource "google_cloudfunctions_function" "main" {
   project               = var.project_id
   region                = var.region
   service_account_email = var.service_account_email
+
+
+  dynamic "timeouts" {
+    for_each = var.operation_timeouts
+    content {
+      create = timeouts.value["create"]
```

Comment:
We may have to a lookup here for default value.

Reply:
Added optional in variable. it's an experimental feature. Please let me know if you any other suggestions.